src/EconomicStrategy/EconomicStrategyHelpers.ts

```diff
@@ -25,7 +26,55 @@ const exceedsMaxDeposit = (txRequest: any, economicStrategy: IEconomicStrategy):
  */
 const isAboveMinBalanceLimit = async (config: Config): Promise<boolean> => {
```

Comment:
This method should have account parameter, otherwise, you add some potential racing conditions.

Reply:
Can you describe this?

Reply:
Is it then preferable to put it here https://github.com/ethereum-alarm-clock/timenode-core/blob/master/src/Router/Router.ts#L54 and pass it down?? I'm not sure that would eliminate a possible race condition since this has to execute sync  before an action takes place

Reply:
If the next account has changed after isAboveMinBalanceLimit then this result is invalid.

Reply:
So we need to pass the account `next` to the Actions too

Reply:
That's a good refactor